### ISSUE TITLE
autoformatting: if-then-else chaining increases indentation

### ISSUE THREAD

**Issue by shwestrick** [shwestrick/parse-sml#29]
The autoformatter should handle chains of if-then-else expressions to make sure that indentation doesn't get
funked up, like in the example below.

INPUT
```sml
val x =
  if b1 then e1
  else if b2 then e2
  else if b3 then e3
  else if b4 then e4
  else if b5 then e5
  else if b6 then e6
  else if b7 then e7
  else e8
```

CURRENT OUTPUT
```sml
val x =
  if b1 then
    e1
  else
    if b2 then
      e2
    else
      if b3 then
        e3
      else
        if b4 then
          e4
        else
          if b5 then
            e5
          else
            if b6 then e6 else if b7 then e7 else e8
```

**Comment by shwestrick** [comment]
Some progress on this in 6c468d55bfdaf63f84f0789ba361ef819e46a912

Added a combinator called `besideAndAbove` which is like `beside`, except it causes the second doc to use the indentation of the first.

```
    d1       d2           besideAndAbove (d1, d2)
  .......   ****             .......
  .......   ***    ====>     .......
  .....                      .....****
                             ***
```

Using `besideAndAbove` to attach the else-branch of a conditional, we get formatting like this:

```sml
val x =
  if b1 then
    e1
  else if b2 then
    e2
  else if b3 then
    e3
  else if b4 then
    e4
  else if b5 then
    e5
  else if b6 then e6 else if b7 then e7 else e8
```

This feels like a step in the right direction. To fix the weird final line of the above example, it seems like we would need to traverse the children of an if/then/else to find the whole chain, before formatting.

**Comment by shwestrick** [comment]
Although I suppose the irony here is that, if we traverse the children in the CST to identify the whole chain, then it's possible to implement the desired formatting in terms of the old combinators 🥲

**Comment by shwestrick** [comment]
Fixed by #34 :)
